src/webgpu/api/validation/encoding/beginComputePass.spec.ts

```diff
@@ -106,6 +106,40 @@ g.test('timestampWrites,query_set_type')
     t.tryComputePass(isValid, descriptor);
   });
 
+g.test('timestamp_query_feature')
+  .desc(
+    `
+  Test that a validation error is generated if the features doesn't contain 'timestamp-query'
+  feature.
+  `
+  )
+  .params(u => u.combine('featureContainsTimestampQuery', [false, true]))
+  .beforeAllSubcases(t => {
+    const { featureContainsTimestampQuery } = t.params;
+
+    const requiredFeatures: GPUFeatureName[] = [];
+    if (featureContainsTimestampQuery) {
+      requiredFeatures.push('timestamp-query');
+    }
+
+    t.selectDeviceOrSkipTestCase({ requiredFeatures });
+  })
+  .fn(async t => {
+    const { featureContainsTimestampQuery } = t.params;
+
+    const timestampWrite = {
+      querySet: t.device.createQuerySet({ type: 'timestamp', count: 1 }),
```

Comment:
This `createQuerySet` call is supposed to fail if the feature timestamp-query feature is not enabled. That is already tested, in `api/validation/capability_checks/features/query_types.spec.ts`

Since it's impossible to create a timestamp query set without the feature, it's only possible to test `beginComputePass` with:
- `timestampWrites` containing a query set of the wrong type (tested in `timestampWrites,query_set_type`)
- `timestampWrites` containing an invalid query set (not sure if we have a test for this)

Reply:
Ah, right. It's likely I missed checking them. Now I'm thinking that we've not tested the case that timestampWrites contain an invalid query set in calling of begingComputePass yet. So I update the test to check it. wdyt?